byzcoin/service.go

```diff
@@ -1261,13 +1274,15 @@ func (s *Service) updateTrieCallback(sbID skipchain.SkipBlockID) error {
 		log.Lvlf4("%v updating trie for block %d refused, current trie block is %d", s.ServerIdentity(), sb.Index, trieIndex)
 		return nil
 	} else if sb.Index > trieIndex+1 {
-		if s.catchingUp {
-			log.Warn(s.ServerIdentity(), "Got new block while catching up - ignoring block for now")
-			return nil
-		}
+		log.Warn(s.ServerIdentity(), "Got new block while catching up - ignoring block for now")
+		go func() {
+			// This new block will catch up at the end of the current catch up (if any)
+			// and be ignored if the block is already known.
+			s.catchingLock.Lock()
+			s.catchUp(sb)
```

Comment:
does this need a `s.working.Add(1)` to make sure there are no leaky go-routines?